keps/sig-network/20181024-service-topology.md

```diff
@@ -226,6 +244,7 @@ In order to handle headless services, the DNS server needs to know the node corr
 - kube-proxy is updated to watch `PodLocator` for topology filtering of endpoints.
 - coredns watches `PodLocator` objets and performs topology-aware headless services.
 - E2E tests exist for service topology.
+- either one of the fields `topologyKeys` or `internalTrafficPolicy` (or a new field) is chosen to implement Service Topology. The other field should be removed before beta.
```

Comment:
Do we have any examples where we have implemented 2 ways, knowing that one would be removed?

Reply:
Since `topologyKeys` is still alpha, would it be possible to deprecate that field? It seems like we shouldn't add a new field here without a plan for deprecating the other field. I'd imagine that writing implementations for all potential variations of both these fields would get complex.

Reply:
> Since topologyKeys is still alpha, would it be possible to deprecate that field? It seems like we shouldn't add a new field here without a plan for deprecating the other field. I'd imagine that writing implementations for all potential variations of both these fields would get complex.

This is a good point -- the kube-proxy implementation would get weird with both fields at play. I guess one clean alternative we can use so that we can keep both is to have separate feature gates for `topologyKeys` and `internalTrafficPolicy`.

Reply:
Updated the KEP to propose a new feature gate `ServiceInternalTrafficPolicy` so we can turn on / off `topologyKeys` and `internalTrafficPolicy` independantly.